formatter.go

```diff
@@ -14,33 +15,62 @@ import (
 // the write entry
 type Formatter interface {
 	Format(w io.Writer, logLevel LogLevel, msg string, timestamp time.Time, fields Fields) error
+	AddMapping(fieldMapptin FieldMapping)
 }
 
 // JSONFormatter used to output logs in JSON format
 type JSONFormatter struct {
-	Compact bool
+	FieldMapping FieldMapping
+	Compact      bool
 }
 
 func (j JSONFormatter) getKey(key string) string {
 	if j.Compact {
-		return "@" + key[:1]
+		var mappedKey string
+		var ok bool
+
+		if mappedKey, ok = j.FieldMapping[key]; !ok {
+			mappedKey = key
+		}
+
+		return mappedKey
 	}
 	return key
 }
 
+func (j *JSONFormatter) AddMapping(fieldMapping FieldMapping) {
+	var mapping = make(FieldMapping, 3)
+
+	// first add the custom mappings
+	for k, v := range fieldMapping {
+		if strings.HasPrefix(v, "@") {
+			fmt.Fprintf(os.Stderr, "value cannot be prefixed with @: %s", v)
+			os.Exit(1)
```

Comment:
Do we really want to exit the service here?

Reply:
That's a good point. 
The thought here was how other libs and frameworks work, if I would do something that is not right would raise, for example, if I have a web framework that I try to start specifying a port that is already taken in the system.
Maybe this is not as serious that I want to prevent the application using the logger stop, perhaps just showing a warning and ignoring that field?
The reason I added that check was to prevent key collision with the built-in keys `@t`, `@m` and `@l`

Reply:
In a non-microservice oriented design I would agree with @dfurtado. However, this will make the service restart over and over again. It's better to log this as an error and catch it using our logging tools

Reply:
Fixed!